### DISCUSSION TITLE
How does outline index documents and how does the search work

### DISCUSSION THREAD

**Discussion by sevaho** [outline/outline#2630] [Self Hosting]
I've made a document on outline version v0.58.2 with title 'Technotronic music', the content of the document contains only the words 'Technotronic music'. I published the document for everyone.
However I cannot see it in the search results when I search for it.

Document:

![image](https://user-images.githubusercontent.com/10941956/135997869-b28eb431-a799-47a9-964d-fb95ff805e2b.png)

No results:

![image](https://user-images.githubusercontent.com/10941956/135997779-4a9f0150-02b9-4bcb-b6bf-f03b7084f75e.png)

**Comment by tommoor** [comment]
What type of database are you using? The search uses standard Postgres vector indexes… you should see the column for the index alongside the document in the db.

**Reply by sevaho** [reply]
We are using AWS Aurora postgres v12.

**Reply by sevaho** [reply]
We have moved from Google DB to AWS aurora via pg_dump and pg_restore could this be the culprit?

**Reply by SimonVHB** [reply]
Hi, I'm a colleague of sevaho.
When I have a look in our database, I find that the searchVector column is empty (null) for 68 of our documents (including the technotronic music example from above).
This will probably be the culprit?

**Reply by tommoor** [reply]
Likely the columns are empty on documents created since you moved database as the dump will not have moved the stored triggers and functions for creating the index. 

https://github.com/outline/outline/blob/a134773d4e79b2b1f9126c3a13e72479335c5d86/server/migrations/20160711071958-search-index.js#L16-L17

https://github.com/outline/outline/blob/main/server/migrations/20210716071454-search-index-previousTitles.js#L6-L15

**Reply by SimonVHB** [reply]
Thanks for all the info!
We finally found the problem, our database was still in replication mode due to the migrations we were doing, which disabled triggers...
